providers/org/datacite/normalizer.py

```diff
@@ -0,0 +1,585 @@
+from share.normalize.utils import format_doi_as_url
+
+
+logger = logging.getLogger(__name__)
+
+THE_REGEX = re.compile(r'(^the\s|\sthe\s)')
+DATE_REGEX = re.compile(r'(^[^a-zA-Z]+$)')
+
+
+def force_text(data):
+    if isinstance(data, str):
+        return data
+    elif isinstance(data, dict):
+        if '#text' in data:
+            return data['#text']
+        raise Exception('#text is not in {}'.format(data))
+    elif isinstance(data, list):
+        for item in data:
+            if isinstance(item, str):
+                return item
+        raise('No value in list {} is a string.'.format(data))
+    else:
+        raise Exception('{} is not a string or a dictionary.'.format(data))
+
+
+class Link(Parser):
+    url = tools.RunPython(
+        'format_link',
+        tools.RunPython(
+            force_text,
+            ctx
+        )
+    )
+    type = tools.Static('doi')
+
+    def format_link(self, link):
+        return format_doi_as_url(self, link)
+
+
+class AlternateLink(Parser):
+    schema = 'Link'
+
+    url = tools.RunPython(
+        force_text,
+        ctx
+    )
+    type = tools.Static('misc')
+
+
+class RelatedLink(Parser):
+    schema = 'Link'
+
+    url = tools.RunPython(
+        force_text,
+        ctx
+    )
+    type = tools.RunPython('lower', tools.Try(ctx['@relatedIdentifierType']))
+
+    def lower(self, type):
+        return type.lower()
+
+
+class ThroughLinks(Parser):
+    link = tools.Delegate(Link, ctx)
+
+
+class ThroughAlternateLinks(Parser):
+    schema = 'ThroughLinks'
+
+    link = tools.Delegate(AlternateLink, ctx)
+
+
+class ThroughRelatedLinks(Parser):
+    schema = 'ThroughLinks'
+
+    link = tools.Delegate(RelatedLink, ctx)
+
+
+class Affiliation(Parser):
+    pass
+
+
+class ContributorInstitution(Parser):
+    schema = 'Institution'
+
+    name = ctx.contributorName
+
+    class Extra:
+        contributor_type = tools.Try(ctx.contributorType)
+
+
+class ContributorOrganization(Parser):
+    schema = 'Organization'
+
+    name = ctx.contributorName
+
+    class Extra:
+        contributor_type = tools.Try(ctx.contributorType)
+
+
+class CreatorInstitution(Parser):
+    schema = 'Institution'
+
+    name = ctx.creatorName
+
+
+class CreatorOrganization(Parser):
+    schema = 'Organization'
+
+    name = ctx
+
+
+class Identifier(Parser):
+    base_url = tools.Try(ctx['@schemeURI'])
+    url = tools.Join(tools.Concat(tools.Try(ctx['@schemeURI']), tools.Try(ctx['#text'])), joiner='/')
+
+
+class ThroughIdentifiers(Parser):
+    identifier = tools.Delegate(Identifier, ctx)
+
+
+class ContributorPerson(Parser):
+    schema = 'Person'
+
+    suffix = tools.ParseName(ctx.contributorName).suffix
+    family_name = tools.ParseName(ctx.contributorName).last
+    given_name = tools.ParseName(ctx.contributorName).first
+    additional_name = tools.ParseName(ctx.contributorName).middle
+    identifiers = tools.Map(tools.Delegate(ThroughIdentifiers), tools.Concat(tools.Try(ctx, 'nameIdentifier')))
```

Comment:
If there is only one `nameIdentifier` then `tools.Try(ctx, 'nameIdentifier')` returns a string.